src/com/google/cose/Ec2SigningKey.java

```diff
@@ -114,6 +118,74 @@ public ECPublicKey getPublicKey() {
     return (ECPublicKey) this.keyPair.getPublic();
   }
 
+  private static byte[] arrayFromBigNum(BigInteger n, int curveSize) {
+    byte[] rgb = new byte[(curveSize + 7) / 8];
+    byte[] rgb2 = n.toByteArray();
+    if (rgb.length == rgb2.length) {
+      return rgb2;
+    }
+    if (rgb2.length > rgb.length) {
+      System.arraycopy(rgb2, rgb2.length - rgb.length, rgb, 0, rgb.length);
+    } else {
+      System.arraycopy(rgb2, 0, rgb, rgb.length - rgb2.length, rgb2.length);
```

Comment:
This shouldn't happen. The `rgb.length == rgb2.length` check would preclude this else statement occurring. If it did occur, that should throw an error instead.